File: arrays/removeDuplicates/removeDuplicates.py
## Slow Pointer Fast Pointer approach 

class Solution:
    def removeDuplicates(self, nums):
        if not nums:
            return 
        slow = fast = 0
        while fast <= len(nums) - 1:
            if nums[fast] != nums[slow]:
                nums[slow+1] = nums[fast]
                slow += 1
            fast += 1
        return slow + 1

File: arrays/removeDuplicates/removeDuplicates.java
// Accepted Solution
// So this solution works for sorted arrays.

class Solution {
    public int removeDuplicates(int[] nums) {
    int i = nums.length > 0 ? 1 : 0;  // This one makes sure that if nums.length == 0 we dont process it
    for (int n : nums) // Enhanced for loop
        if (n > nums[i-1])
            nums[i++] = n; // pushing all the elements ahead
    return i;
    }
}